### PR TITLE
Add node-sass-json-importer, device-state-tracker and breakpoint-sass

### PR DESCRIPTION
**Pull request by ThaNarie**
This PR adds and updates how we work with variables and mediaQueries in SCSS.

### 1. breakpoint-sass

http://breakpoint-sass.com/
Similar to our `respond-to` mixins, but with some additional features, and already maintained by others.

The biggest difference is that you pass actual media-query values (or shortcuts), instead of a name from a map.

### 2. node-sass-json-importer

https://www.npmjs.com/package/node-sass-json-importer
Allows you to import `.json` files from scss. Replaces https://www.npmjs.com/package/sassify-webpack-plugin which has its own issues and is fit for replacement.

Using node-sass-json-importer you can share variables between JS and SCSS. Every value in a json object will be made available as SCSS variable; arrays will be lists, objects will be maps.

The only downside using such a mechanism is that your variables are not available in SCSS itself, so will not provide autocompletion or 'go to declaration'. This could be solved by re-defining the variables in SCSS after importing them, but that results in code duplication, making changes cumbersome.

An example could be then:
```json
{
  "colors": {
    "hover": "#D82E18"
  }
}
```
```scss
import '../variables/colors.json';
$colorHover: map-get($colors, hover);
```

This could also be useful to store colors, spacings, and other values you would like to share between JS and SCSS.

### 3. seng-device-state-tracker

https://github.com/mediamonks/seng-device-state-tracker
A thin wrapper around the matchMedia API, allowing you to check the current breakpoint and watch for changes, making it easy to keep the UI sync between CSS and JS logic that react on screen sizes.

It's exported as a 'singleton' from `util/deviceStateTracker.ts`, and will only be bundled when imported somewhere. Configuration happens from an enum and from the shared media-query values from the JSON files.

### CODE REVIEW

**Comment by larsvanbraam** [comment]
Not sure if I'm a big fan of manually re-typing all the scss variables. But I really like autocompletion on my variables, so if there is no other way of getting autocompletion on the variables we could map the breakpoints to scss variables by default. This way you can start using them out of the box, since a lot of times (at least in my latest projects) the only variables you share between js and scss are the breakpoints.

**Comment by ThaNarie** [comment]
Yeah, there are a few things we would like to have, but they don't work nice together:

1. share stuff between JS and TS
2. having autocomplete on variables (which cannot be done without generating additional files/vars)
3. don't generate additional (duplicate) files with the same values, just for autocompletion

re-defining maps to variables might be the nicest middle road. And you still have the option to use the map directly somewhere.

Previously the sassify-webpack-plugin generated an additional SCSS file from a TS file, but this wasn't a super nice solution:
* 'duplicate' file in git
* changes to this file are overwritten on next run, so it's not super clear where to change these values
* generating files in webpack that trigger another webpack compile can result in multiple re-compiles on a single change.

Ideally we'd be looking for a "definition file for scss variables" outside of the source root (just for your IDE), that can be auto-generated (without having a weird watcher setup).

**Comment by jesse-mm** [comment]
I've already talked about it with @ThaNarie. Ut would be a good idea to have some CLI tool that just generates it from a JSON file. That just (for example) dumps this scss file in node_modules (which is sort-of auto-ignored). So you still can reference the variables.

File: src/app/data/variables/media-queries.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (jesse-mm) [inline src/app/data/variables/media-queries.json]: directory should have singular name `variable`

**Comment by ThaNarie** [comment]
@ReneDrie @larsvanbraam @riccomediamonks @pigeonfresh @jesse-mm Can you have another look, is now similar to Vue Skeleton.

**Comment by ThaNarie** [comment]
@larsvanbraam It's not the media-queries folder, it's the folder where json files with shared variables are put in.

**Comment by larsvanbraam** [comment]
@ThaNarie So maybe naming it `shared-variable` would be more descriptive?

**Comment by ReneDrie** [comment]
It looks good. The only thing I think is a bit overkill, is the list of MediaQueries
`    "XSMALL": "'(max-width: 479px)'",
    "SMALL": "'(min-width: 480px)'",
    "MEDIUM": "'(min-width: 768px)'",
    "LARGE": "'(min-width: 1024px)'",
    "XLARGE": "'(min-width: 1280px)'",
    "XXLARGE": "'(min-width: 1440px)'",
    "XXXLARGE": "'(min-width: 1680px)'"`

To me the below is fine:
`    "XSMALL": "'(max-width: 479px)'",
    "SMALL": "'(min-width: 480px)'",
    "MEDIUM": "'(min-width: 768px)'",
    "LARGE": "'(min-width: 1024px)'"`

We have something like this in Vue, and it's always to get an idea

**Pull request merged**